stl/inc/format

```diff
@@ -1116,6 +1116,47 @@ _OutputIt _Write(_OutputIt _Out, monostate) {
     return _Out;
 }
 
+// This size is derived from the maximum length of an arithmetic type. The two contenders for widest are double and long
+// long. long long has a max length of ceil(log_10(2^64)) = 20 characters. double has a max length of
+// limits<double>::max_digits10 + the decimal + the sign + e + the exponent's sign + ceil(log_10(DBL_MAX_10_EXP))
+// = 17 + 1 + 1 + 1 + 3 = 24. An example is DBL_MAX which is "1.7976931348623158e+308".
+inline constexpr size_t _Format_min_buffer_length = 24;
+
+// clang-format off
+template <class _CharT, class _OutputIt, class _Arithmetic>
+    requires (is_arithmetic_v<_Arithmetic> && !same_as<_Arithmetic, bool> && !same_as<_Arithmetic, _CharT>)
+_OutputIt _Write(_OutputIt _Out, _Arithmetic _Value) {
+    // clang-format on
+    // TODO: Reusable buffer
+    array<char, _Format_min_buffer_length> _Buffer;
+    auto [_End, _Ec] = _STD to_chars(_Buffer.data(), _Buffer.data() + _Buffer.size(), _Value);
+    _STL_ASSERT(_Ec == errc{}, "to_chars failed");
+    return _STD copy(_Buffer.data(), _End, _Out);
+}
+
+template <class _CharT, class _OutputIt>
+_OutputIt _Write(_OutputIt _Out, bool _Value) {
+    return _Write(_Out, _Value ? "true" : "false");
+}
+
+template <class _CharT, class _OutputIt>
+_OutputIt _Write(_OutputIt _Out, _CharT _Value) {
+    *_Out = _Value;
+    return ++_Out;
+}
+
+template <class _CharT, class _OutputIt>
+_OutputIt _Write(_OutputIt _Out, const void* _Value) {
+    // TODO: Reusable buffer
+    array<char, 16> _Buffer;

```

Comment:
I think we should also use `_Format_min_buffer_length` (those 8 chars wont probably hurt) 

Or we could add `_Format_min_buffer_length_pointer` and specialize that for 32biut vs 64bit?

Reply:
Let's reuse the other size, I don't really expect this code to make it to the finished product anyway.

Reply:
Nothing is as permanent as the interim